incubator/virtualcluster/pkg/syncer/resources/configmap/checker.go

```diff
@@ -55,97 +54,74 @@ func (c *controller) PatrollerDo() {
 		return
 	}
 
-	wg := sync.WaitGroup{}
-	numMissMatchedConfigMaps = 0
-
-	for _, clusterName := range clusterNames {
-		wg.Add(1)
-		go func(clusterName string) {
-			defer wg.Done()
-			c.checkConfigMapsOfTenantCluster(clusterName)
-		}(clusterName)
-	}
-	wg.Wait()
-
 	pConfigMaps, err := c.configMapLister.List(labels.Everything())
 	if err != nil {
 		klog.Errorf("error listing configmaps from super master informer cache: %v", err)
 		return
 	}
+	pSet := differ.NewDiffSet()
+	for _, pCM := range pConfigMaps {
+		pSet.Insert(differ.ClusterObject{Object: pCM})
+	}
 
-	for _, pConfigMap := range pConfigMaps {
-		clusterName, vNamespace := conversion.GetVirtualOwner(pConfigMap)
-		if len(clusterName) == 0 || len(vNamespace) == 0 {
+	vSet := differ.NewDiffSet()
+	for _, cluster := range clusterNames {
+		listObj, err := c.multiClusterConfigMapController.List(cluster)
```

Comment:
there is a race. if some vc is deleted during this time, the vSet is missing some objects. patrol will requeue them, it's fine